src/Reflection/ClassReflection.php

```diff
@@ -757,26 +763,28 @@ public function getParents(): array
 	}
 
 	/**
+	 * @param (callable(ClassReflection): bool)|null $filterCallback
+	 *
 	 * @return ClassReflection[]
 	 */
-	public function getInterfaces(): array
+	public function getInterfaces(?callable $filterCallback = null): array
 	{
 		if ($this->cachedInterfaces !== null) {
 			return $this->cachedInterfaces;
```

Comment:
I almost merged this but then I realized the fatal flaw here. You're filtering the interfaces and parent class, and regardless of what gets filtered, you cache that result. And when the result is already cached, you return the cached result regardless of the filter callback.

Reply:
There is two things:

> You're filtering the interfaces and parent class, and regardless of what gets filtered, you cache that result.

This was a big flaw indeed ; I should not cache the result when a filterCallback is used.
Nice catch.

> And when the result is already cached, you return the cached result regardless of the filter callback.

Here, I was hesitating and I'd like to have your point of view.

Maybe the `filterCallback` name is not the best, the idea was the following:
- If the result is cached, use it because it require no computation
- If the result is not cached, use the filterCallback to early return null/less class/interface in order to avoid extra computation which are useless (like resolving parentClass/interface without the constant I look for)
- If the full result is computed without filter, cache it

So the filterCallack is not a way to guarantee less results, it a way to guarantee less computations.

I could change to
```
if ($this->cachedInterfaces !== null && $filterCallback === null) {
    return $this->cachedInterfaces;
}
```
but this would means that every time I run `resolveParentPhpDocBlocks` I would call 
```
$classReflection->getParentClass($filterCallback)
```
and not use the cache, ending with lower performance for PHPStan tool. 

Would it be better with another name than `filterCallback` ?

Reply:
I feel like I don't want to make ClassReflection API to be more complicated and potentially slower just for some bug avoidance in PhpDocBlock.

I'm gonna try something, wait a moment.